.github/workflows/eden.yml

```diff
@@ -22,7 +22,43 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
+  check_pr_publish:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check Image Availability
+        run: |
+          DOCKER_HUB_PATH="evebuild/danger"
+          # currently we only test amd64 (kvms and xen) with Eden
+          IMAGE_AMD64_XEN="pr${{ github.event.pull_request.number }}-xen"
+          IMAGE_AMD64_KVM="pr${{ github.event.pull_request.number }}-kvm"
+
+          amd64_xen_res=$(curl -s "https://hub.docker.com/v2/repositories/${DOCKER_HUB_PATH}/tags/${IMAGE_AMD64_XEN}")
+          amd64_kvm_res=$(curl -s "https://hub.docker.com/v2/repositories/${DOCKER_HUB_PATH}/tags/${IMAGE_AMD64_KVM}")
+          if [[ "$(echo "$amd64_xen_res" | jq -r '.name')" == "${IMAGE_AMD64_XEN}" && "$(echo "$amd64_kvm_res" | jq -r '.name')" == "${IMAGE_AMD64_KVM}" ]]; then
+              echo "Docker image ${DOCKER_HUB_PATH}:${IMAGE_AMD64_XEN} and ${DOCKER_HUB_PATH}:${IMAGE_AMD64_KVM} are available on Docker Hub."
+              exit 0
+          else
+              echo "Docker image ${DOCKER_HUB_PATH}:${IMAGE_AMD64_XEN} and ${DOCKER_HUB_PATH}:${IMAGE_AMD64_KVM} are not available on Docker Hub."
+              exit 1
+          fi
+
+  report_pr_publish:
+    needs: check_pr_publish
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/github-script@v6
+        if: failure()
+        with:
+          script: |
+            github.rest.issues.createComment({
```

Comment:
I think to leave a comment, the step requires write-permissions. And I'm not sure it will have it in the case of a PR created from a forked repo (our usual workflow).
But we can test. The permissions part is still magic to me.

Reply:
it implicitly uses GITHUB_TOKEN.

Reply:
Yep, but the token's permission may not be enough in this case. They can be pretty restrictive when running PRs from forked repos.

Reply:
I don't have access but here is some info 
https://docs.github.com/en/actions/security-guides/automatic-token-authentication#permissions-for-the-github_token

Reply:
In my case, this fix did not work... 
I had to change the trigger of my workflow to pull_request_target. And adopt the script to work from the "lf-edge" version of the code (some extra fetches were necessary). But here the trigger is another, so I'm not sure how to fix it...